src/services/order.ts

```diff
@@ -22,27 +29,45 @@ export class OrderService {
     order: OrderModel,
     orderRequest: Omit<OrderRequest, 'timestamp'>
   ) {
-    for (const productId in orderRequest.items) {
-      const product = await this.productModel.findOneById(productId);
+    for (const orderItem of orderRequest.items) {
+      const product = await this.productModel.findOneById(orderItem.productId);
+
+      const additives = [];
+
+      for (const additiveId in orderItem.additives) {
+        const additive = await this.additiveModel.findOneById(additiveId);
+        if (additive) {
+          additives.push({
+            id: additiveId,
+            name: additive.name,
+            price: additive.price,
+            count: orderItem.additives[additiveId]
+          });
+        } else {
+          throw new Error(`Additive with id: ${additiveId} not found`);
```

Comment:
I think it's better to replace Error with ValidationError to have 400 status code instead of 500